src/lib/datepicker/datepicker-input.ts

```diff
@@ -123,49 +133,45 @@ export class MatDatepickerInput<D> implements ControlValueAccessor, OnDestroy, V
 
   /** The value of the input. */
   @Input()
-  get value(): D | null { return this._value ? this._value.asDate() : null; }
+  get value(): D | null { return this._value ? this._value.getFirstSelectedDate() : null; }
   set value(value: D | null) {
-    const oldDate = this._value? this._value.clone(): null;
-    if (this._value) {
+    const oldDate = this._value.getFirstSelectedDate();
+
+    if (this._value && !this._dateAdapter.sameDate(value, this._value.getFirstSelectedDate())) {
       this._value.add(value);
     } else {
       this._value = new MatSingleDateSelection(this._dateAdapter, value);
```

Comment:
The problem is that the input wouldn't have one if it's not connected to a datepicker. Which I guess should never be the case since it's the datepicker input, but one could technically not pass a reference to a `matDatepicker`. I will opt for just throwing an error when one tries to set the `value` but it has not associated `MatDatePicker`